tests/unit/src/test/scala/tests/codeactions/CompanionObjectSuite.scala

```diff
@@ -0,0 +1,478 @@
+  )
+
+  check(
+    "insert companion object of trait inside parent object",
+    """|object Baz {
+       |  trait F<<>>oo {
+       |
+       |  }
+       |
+       |  class Bar {
+       |
+       |  }
+       |}
+       |""".stripMargin,
+    s"""|${CreateCompanionObjectCodeAction.companionObjectCreation}
+        |""".stripMargin,
+    """|object Baz {
+       |  trait Foo {
+       |
+       |  }
+       |
+       |  object Foo {
+       |
+       |  }
+       |
+       |  class Bar {
+       |
+       |  }
+       |}
+       |""".stripMargin
+  )
+
+  check(
+    "insert companion object of trait inside parent class",
+    """|class Baz {
+       |  trait F<<>>oo {
+       |
+       |  }
+       |
+       |  class Bar {
+       |
+       |  }
+       |}
+       |""".stripMargin,
+    s"""|${CreateCompanionObjectCodeAction.companionObjectCreation}
+        |""".stripMargin,
+    """|class Baz {
+       |  trait Foo {
+       |
+       |  }
+       |
+       |  object Foo {
+       |
+       |  }
+       |
+       |  class Bar {
+       |
+       |  }
+       |}
+       |""".stripMargin
+  )
+
+  check(
+    "don't create existing companion object of trait",
+    """|trait Fo<<>>o{
+       |
+       |}
+       |
+       |object Foo {
+       |
+       |}
+       |""".stripMargin,
+    s"""|${ExtractRenameMember.title("trait", "Foo")}
+        |${CreateCompanionObjectCodeAction.companionObjectInfo}
+        |""".stripMargin,
+    """|trait Foo{
+       |
+       |}
+       |
+       |object Foo {
+       |
+       |}
+       |""".stripMargin,
+    selectedActionIndex = 1
+  )
+
+  check(
+    "don't create existing companion object of trait inside a parent object",
+    """|object Bar{
+       |  trait Fo<<>>o{
+       |
+       |  }
+       |
+       |  object Foo {
+       |
+       |  }
+       |
+       |  object Baz{
+       |
+       |  }
+       |
+       |}
+       |""".stripMargin,
+    s"""|${CreateCompanionObjectCodeAction.companionObjectInfo}
+        |""".stripMargin,
+    """object Bar{
+      |  trait Foo{
+      |
+      |  }
+      |
+      |  object Foo {
+      |
+      |  }
+      |
+      |  object Baz{
+      |
+      |  }
+      |
+      |}
+      |""".stripMargin
+  )
+
+  check(
+    "don't create existing companion object of trait inside a parent package",
+    """|package bar
+       |  trait Fo<<>>o{
+       |
+       |  }
+       |
+       |  object Foo {
+       |
+       |  }
+       |
+       |  object Baz{
+       |
+       |  }
+       |
+       |
+       |""".stripMargin,
+    s"""|${ExtractRenameMember.title("trait", "Foo")}
+        |${CreateCompanionObjectCodeAction.companionObjectInfo}
+        |""".stripMargin,
+    """|package bar
+       |  trait Foo{
+       |
+       |  }
+       |
+       |  object Foo {
+       |
+       |  }
+       |
+       |  object Baz{
+       |
+       |  }
+       |
+       |
+       |""".stripMargin,
+    selectedActionIndex = 1
+  )
+
+  check(
```

Comment:
@ckipp01 yes, sure I would squash things! But there is a huuuuuge problem with the current state of the code: it doesn’t seem to parse the braceless syntax of the test! Can you please take a look?

Reply:
Honestly my personal opinions is to just get the feature working and not worry about braceless for now. Most of the time scalafmt can just take care of this, and it seems like the vast majority of the complexity in the pr is around braceless syntax. Don't be afraid to push the feature without support for braceless, and then we can iterate and try to improve or add it later. It's up to you though.

Reply:
@ckipp01  thank you very much for the input! 
But the biggest problem with the braceless is that it does not get parsed in the first place to get to the point of the execution of the code action! 
I think this issue demands some more attention!

Reply:
Even copying and pasting from [Scastie](https://scastie.scala-lang.org/w7MswamuT4eMEBFSq2XkLA) into the test, makes the whole tree to be `None` still! nothing gets parsed!

Reply:
Ok, now actually in the live demo, the braceless syntax works perfectly with the current code! I wonder what is wrong with the test case! I think I should just remove the test for braceless, then???
@tgodzik

Reply:
I think most probably the test method is messing up with white space, behind the scenes, somewhere🧐

Reply:
huh, got it! The whole problem is coming from here in the `check` method of `BaseCodeActionLspSuite`
 initialize(s"""/metals.json
                           |{"a":{$scalacOptionsJson "scalaVersion" : "$scalaVersion"}}
                           |$scalafixConf
                           |/$path
                           |$fileContent"""
          .stripMargin
        )

when it strips the margin, it becomes wellformed json with the braceless syntax somehow; but when it does not, due to the `:` notation of the bracelsss syntax, it somehow gets mistaken as malformed json!

Reply:
according to @tgodzik 

> test creating companion object with Scala 3 and optional braces (this one needs to be done in Scala3CodeActionLspSuite since unit tests are able to only test the 2.13.8 version)